File: Ejemplo_en_C#/Programa_Video_Juego/Programa_Video_Juego/Clases/Pistola.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Programa_Video_Juego.Clases
{
    class Pistola : Arma
    {
        private float cargador_rapido;

        public Pistola(int municion, float tiempo_recarga, int daño) : base(municion, tiempo_recarga, daño)
        {
            cargador_rapido = 1.5f;
            Municion = municion;
            Tiempo_recarga = tiempo_recarga - cargador_rapido;
            Daño = daño;
        }

        public override string Disparar(Enemigo objetivo)
        {
            if (municion > 0)
            {
                
                if (objetivo.Vida <= 0)
                {
                    objetivo.Vida = 0;
                    return "\nNo se pudo disparar la PISTOLA, el ENEMIGO esta muerto." + "\nVida del objetivo: " + objetivo.Vida + "\nMunicion: " + municion;
                }
                else
                {
                    objetivo.Vida -= daño;
                    municion -= 1;
                }
                
                return "\nSe disparo la PISTOLA" + "\nVida del objetivo: " + objetivo.Vida + "\nMunicion: " + municion;
            }
            else if (municion <= 0)
            {
                return "\nNo se pudo disparar la PISTOLA, no tiene municion." + municion;
            }
            else
            {
                return "Ocurrio un error";
            }
        }
    }
}

File: Ejemplo en C/Programa/Pistola.c
#include <stdlib.h>
#include <string.h>
#include <stdio.h>

#include "Pistola.h"
#include "Enemigo.h"

void __Nombre_Arma_Pistola()
{
    printf("Pistola");
}

void __Disparar_Pistola(void *esta_arma, void *este_enemigo){
    Gun *ref_arma = (Gun*) esta_arma;
    Enemigo *ref_enemigo = (Enemigo*) este_enemigo;
    if (ref_enemigo->vida > 0)
    {
        if (ref_enemigo->vida <= 0){
            ref_enemigo->vida = 0;
            printf("No se pudo disparar el Pistola, el Enemigo ya esta muerto.\nVida del objetivo: %d \nMunicion: %d", ref_enemigo->vida, ref_arma->municion);
        }
        else
        {
            ref_enemigo->vida -= 2;
            ref_arma->municion -= 2;
            printf("Se disparo el Pistola.\nVida del objetivo: %d \nMunicion: %d", ref_enemigo->vida, ref_arma->municion);
        }
        
    }
    else if (ref_arma->municion <= 0)
    {
        printf("No se pudo disparar el Pistola. Sin municion: %d", ref_arma->municion);
    }
    else
    {
        printf("Ocurrio un error");
    }
}

Pistola *Crear_Pistola(int municion, float tiempo_recarga, int dano)
{
    Pistola *nueva_pistola = (Pistola*)malloc(sizeof(Pistola));

    nueva_pistola->arma = *Crear_Arma(municion,tiempo_recarga,dano);
    nueva_pistola->arma.disparar_func = __Disparar_Pistola;
    nueva_pistola->arma.nombre_func = __Nombre_Arma_Pistola;
    nueva_pistola->cargador_rapido = 1.5f;

    nueva_pistola->arma.tiempo_recarga -= nueva_pistola->cargador_rapido;

    return nueva_pistola;
}

void Destruir_Pistola(Pistola *esta_pistola)
{
    Destruir_Arma((Gun*)esta_pistola);
}